src/xlsb.rs

```diff
@@ -188,36 +188,48 @@ impl<RS: Read + Seek> Xlsb<RS> {
         };
         let mut buf = vec![0; 1024];
         let mut number_formats = BTreeMap::new();
+        let mut xfs = Vec::new();
 
-        // Search for BrtBeginFmts
-        if iter.next_skip_blocks(0x0267, &[], &mut buf).is_ok() {
-            let len = read_usize(&buf);
-
-            for _ in 0..len {
-                let _ = iter.next_skip_blocks(0x002C, &[], &mut buf)?; // BrtFmt
-                let fmt_code = read_u16(&buf);
-                let fmt_string = wide_str(&buf[2..], &mut 0)?.into_owned();
-                number_formats.insert(fmt_code, fmt_string);
-            }
-
-            // Search for BrtBeginCellXFs
-            if iter.next_skip_blocks(0x0269, &[], &mut buf).is_ok() {
-                let len = read_usize(&buf);
+        loop {
+            match iter.read_type()? {
+                0x0267 => {
+                    // BrtBeginFmts
+                    let _len = iter.fill_buffer(&mut buf)?;
+                    let len = read_usize(&buf);
 
-                for _ in 0..len {
-                    let _ = iter.next_skip_blocks(0x002F, &[], &mut buf)?; // BrtXF
-                    let num_fmt = read_u16(&buf[2..4]);
-                    self.formats.push(match number_formats.get(&num_fmt) {
-                        Some(fmt) if is_custom_date_format(fmt) => CellFormat::Date,
-                        None if is_builtin_date_format_id(num_fmt.to_string().as_bytes()) => {
-                            CellFormat::Date
-                        }
-                        _ => CellFormat::Other,
-                    });
+                    for _ in 0..len {
+                        let _ = iter.next_skip_blocks(0x002C, &[], &mut buf)?; // BrtFmt
+                        let fmt_code = read_u16(&buf);
+                        let fmt_string = wide_str(&buf[2..], &mut 0)?.into_owned();
```

Comment:
```suggestion
                        let fmt_str = wide_str(&buf[2..], &mut 0)?;
                        let fmt = if is_custom_date_format(fmt_str) {
                            CellFormat::Date
                        } else {
                            CellFormat::Other
                        };
                        number_formats.insert(fmt_code, fmt);
```

To avoid allocating the fmt as we're only caring if it is a date or not (for the moment).